pkgs/development/python-modules/ffmpeg-python/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, ffmpeg
+, future
+, pytest
+, pytestrunner
+, pytest-mock
+}:
+
+buildPythonPackage rec {
+  pname = "ffmpeg-python";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "kkroening";
+    repo = "ffmpeg-python";
+    rev = version;
+    sha256 = "0mmydmfz3yiclbgi4lqrv9fh2nalafg4bkm92y2qi50mwqgffk8f";
+  };
+
+  buildInputs = [ pytestrunner ];
+  propagatedBuildInputs = [ ffmpeg future ];
```

Comment:
looks likes this shells out to ffmpeg, for this to work as intended, you'll need to patch the call sights so that it resolves to nix-store paths:

example call: https://github.com/kkroening/ffmpeg-python/blob/4cb7d26f55208b01b123f07b13108bf4143794e2/ffmpeg/_run.py#L292
example commit with patching: https://github.com/NixOS/nixpkgs/pull/64551/commits/06960ac6a7b58af84aeccd208b663443b39a6600

Reply:
propagating ffmpeg doesn't really work because it might conflict with the ffmpeg that a users wants to use in their own environment.

Reply:
I've moved ffmpeg to `buildInputs` in both packages using it, but I'm not sure that's where it belongs.

Reply:
it's technically not needed. Nix will know through the patches that the ffmpeg closure will need to be included. "contextual strings" is the nix teerm

Reply:
Alright, removed from `buildInputs`. Everything is patched, I believe.